samples/tests/traits/mod.rs

```diff
@@ -0,0 +1,43 @@
+samples::test! {
+    volumetric;
+    /// ANCHOR: volumetric
+    trait Volumetric {
+        /// Strain energy density
+        fn psi(&self, j: f64) -> f64;
+        /// Derivative of strain energy with respect to $J$
+        fn d_psi(&self, j: f64, b_psi: f64) -> f64;
+        /// The volumetric contribution to the second Piola-Kirchhoff stress is
+        /// a scalar multiplied by $C^{-1}$ where $C = I + 2E$ in terms of the
+        /// Green-Lagrange strain $E$. The derivative of $J$ with respect to $E$
+        /// is $J C^{-1}$. We'll call the volumetric contribution that scalar
+        /// multiple of $C^{-1}$.
+        fn stress(&self, j: f64) -> f64 {
+            let (_, d_psi) = self.d_psi(j, 1.0);
+            d_psi * j
+        }
+    }
+
+    struct Ogden {
+        k: f64,
+    }
+    impl Ogden {
+        pub fn stress_analytic(&self, j: f64) -> f64 {
+            self.k * 0.5 * (j * j - 1.0)
+        }
+    }
+    impl Volumetric for Ogden {
+        #[autodiff(d_psi, Reverse, Active, Active)]
```

Comment:
I'd prefer to use `ActiveOnly`, but
```
error: did not recognize activity ActiveOnly
  --> samples/tests/traits/mod.rs:29:44
   |
29 |         #[autodiff(d_psi, Reverse, Active, ActiveOnly)]
   |                                            ^^^^^^^^^^

[compiler/rustc_builtin_macros/src/autodiff.rs:201:5] "after 190" = "after 190"
```

Reply:
I think this should be written as
```suggestion
        #[autodiff(d_psi, Reverse, Const, Active, Active)]
```
but that gives
```
error[E0425]: cannot find function `psi` in this scope
  --> samples/tests/traits/mod.rs:29:9
   |
29 |         #[autodiff(d_psi, Reverse, Const, Active, Active)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: you might have meant to call the method
   |
29 |         self.#[autodiff(d_psi, Reverse, Const, Active, Active)]
   |         +++++
help: use the `.` operator to call the method `Volumetric::psi` on `&Ogden`
   |
29 -         #[autodiff(d_psi, Reverse, Const, Active, Active)]
29 +         .psi(#[autodiff(d_psi, Reverse, Const, Active, Active)]
   |
```

Reply:
You will need to give one activity per input, that is Duplicated/Const + Active/Const, since self is neither f32 nor f64 (and also behind a reference, so active is twice not supported for now).
I assume that then duplicated is what you want, since you wrote Active, Active.

Reply:
I clearly can't count. Yes, Const was correct, since you need one activity per input/output argument (incl. self, which you could Duplicate).